be/src/service/service_be/lake_service.cpp

```diff
@@ -1375,4 +1376,44 @@ void LakeServiceImpl::vacuum_full(::google::protobuf::RpcController* controller,
     latch.wait();
 }
 
+void LakeServiceImpl::get_tablet_cache_stats(::google::protobuf::RpcController* controller,
+                                             const ::starrocks::GetTabletCacheStatsRequest* request,
+                                             ::starrocks::GetTabletCacheStatsResponse* response,
+                                             ::google::protobuf::Closure* done) {
+    brpc::ClosureGuard guard(done);
+#ifdef USE_STAROS
+    auto cntl = static_cast<brpc::Controller*>(controller);
+    if (request->tablets_size() == 0) {
+        cntl->SetFailed("missing tablet info");
+        return;
+    }
+
+    auto cache_stats_mgr = _tablet_mgr->cache_stats_mgr();
+    std::vector<std::shared_ptr<lake::TabletCacheStatsContext>> ctxs;
+    for (auto& tablet : request->tablets()) {
+        ctxs.push_back(cache_stats_mgr->submit_get_tablet_cache_stats(tablet.tablet_id(), tablet.version()));
+    }
+    Status st = Status::OK();
+    for (auto& ctx : ctxs) {
+        auto result = ctx->get();
```

Comment:
std::future/promise is a blocking operation, this will block the pthread for the bthread, causes brpc process blocking, shall turn to bthread::future implementation.

Reply:
fixed